var/spack/repos/builtin/packages/sio/package.py

```diff
@@ -19,9 +19,28 @@ class Sio(CMakePackage):
     maintainers = ['vvolkl', 'tmadlener']
 
     version('master', branch='master')
+    version('0.0.4', sha256='72e96e6a1cc8dd3641d3e2bb9876e75bf6af8074e1617220da9e52df522ef5c0')
     version('0.0.3', sha256='4c8b9c08480fb53cd10abb0e1260071a8c3f68d06a8acfd373f6560a916155cc')
     version('0.0.2', sha256='e4cd2aeaeaa23c1da2c20c5c08a9b72a31b16b7a8f5aa6d480dcd561ef667657')
 
+    variant('builtin_zlib', default=True,
+            description='Use and statically link against a builtin version of zlib')
```

Comment:
Would it make sense to simplify the interface, remove this variant and always have a dependency on `zlib`? What would be the gain of using the vendored version?

Reply:
The vendored version is a re-implementation that makes use of vectorized SSE and AVX instructions if they are support. I just checked the `root` package, which ships the same re-implementation, but they are also using the spack version, so I think we could do the same here as well. But I will check with the SIO developers again to see if this would incur significant performance penalties.

Reply:
That sounds good to me. In case would it be possible to have this reimplementation as a standalone library, rather than being compiled as a vendored dependency by all the packages that use it?

Reply:
I have run some benchmarks with both options, and at least in the benchmarked use cases, there is quite a difference between the vendored version and the one from `zlib`. Especially writing is roughly 40 % quicker when using the vendored version, for reading there is only a roughly 5 % difference.

I am not sure, whether we should remove the variant completely and "silently" do the static linking internally against the vendored version (this is what is done currently), or if we should at least keep the variant to show that the option exists, and allow to switch this from the outside. It could be that for other use cases the performance hit is less severe

Regarding a standalone library, I would have to investigate and find all the responsible people and see what their opinions about this are. In general, I think it would be a very nice solution, if all involved parties are OK with it. However, I am not sure about the time frame of this.

Reply:
Thanks for reporting back so quickly. Would there be linking or loading issues if any library in `sio` is used in a project that also depends on `zlib`? My worry is about duplicated symbols in binaries.

Reply:
Good question. From what I can tell, this is currently not a problem. I have a few packages that link both against `sio` and `root` (where `root` uses `zlib`) and at least here I haven't run into any problems yet, regardless of which version of zlib I use to build `sio`. However, I can't say if these few packages really exhaust all the possibilities for different symbol usages. But, I think they at least cover all the uses of `sio` within the spack ecosystem.

Reply:
Ok, given that the use of this library seems quite focused on a few packages:
```console
$ spack dependents --transitive sio
acts  dd4hep  edm4hep  podio
```
I'd be fine going forward with this change without further investigation and try to have an external dependency on the modified zlib in following PRs. Thanks for the info!